addons/mail_plugin/tests/test_controller.py

```diff
@@ -46,32 +42,7 @@ def test_get_partner_blacklisted_domain(self):
             },
         )
 
-        first_company_id = result["partner"]["company"]["id"]
```

Comment:
Maybe mention in the commit message that this method was actually not testing a real use case, as IAP would never enrich a blacklisted domain in the first place.

Because currently it feels like you are removing company creation for blacklisted domain, but it's not the case. Company were never created for blacklisted domains.